include/RouteMap.h

```diff
@@ -305,57 +324,119 @@ struct RouteMapPosition {
+    // at each step of the route calculation, in degrees.
+    // This represents the angle away from Start to End bearing (StartEndBearing).
+    // The normal setting is 100 degrees, which speeds up calculations.
+    // If the route needs to go around land, islands or peninsulas, the user can increase the value.
+    // E.g. the boat may have to go in the opposite direction then back to the destination bearing.
+    double MaxDivertedCourse;
+    // The maximum deviation away from the direct route.
+    double MaxCourseAngle;
+    // The maximum angle at each step of the route calculation.
+    double MaxSearchAngle;
+    // The calculated route will avoid a path where the true wind is above this value in knots.
+    double MaxTrueWindKnots;
+    // The calculated route will avoid a path where the gust wind is above this value in knots.
+    double MaxApparentWindKnots;
+
+    // The calculated route will avoid swells larger than this value in meters.
+    // If the grib data does not contain swell information, the maximum swell value is ignored.
+    // If there is no route within the maximum swell value, the route calculation will fail.
+    double MaxSwellMeters;
+    // The calculated route will not go beyond this latitude, as an absolute value.
+    // If the starting or destination position is beyond this latitude, the route calculation will fail.
+    double MaxLatitude;
+    // The penalty time to tack the boat, in seconds.
+    // The penalty time is added to the route calculation for each tack.
+    double TackingTime;
+    // Balance the influence of the wind and the ocean current on the route calculation.
+    double WindVSCurrent;
+    // The minimum safety distance to land, in nautical miles.
+    // The calculated route will avoid land within this distance.
     double SafetyMarginLand;
 
     bool AvoidCycloneTracks;
-    int CycloneMonths, CycloneDays;
+    // Avoid cyclone tracks within ( 30*CycloneMonths + CycloneDays ) days of climatology data.
+    int CycloneMonths;
+    int CycloneDays;
 
     bool UseGrib;
     enum ClimatologyDataType {DISABLED, CURRENTS_ONLY, CUMULATIVE_MAP, CUMULATIVE_MINUS_CALMS, MOST_LIKELY, AVERAGE};
     enum ClimatologyDataType ClimatologyType;
     bool AllowDataDeficient;
-    double WindStrength; // wind speed multiplier
+    double WindStrength; // wind speed multiplier. 1.0 is 100% of the wind speed in the grib.
+
+    // If true, the route calculation will avoid land, outside the SafetyMarginLand.
+    bool DetectLand;
+    // If true, the route calculation will avoid exclusion boundaries.
+    bool DetectBoundary;
+    // If true and grib data contains ocean currents, the route calculation will use ocean current data.
+    bool Currents;
+    // If true, avoid polar dead zones.
+    // If false, avoid upwind course (polar angle too low) or downwind no-go zone (polar angle too high).
+    bool OptimizeTacking;
+    bool InvertedRegions;
+    bool Anchoring;
```

Comment:
I'm unclear how `InvertedRegions` and `Anchoring' parameters are used.

Reply:
anchoring was never very well implemented but the idea is sometimes the boat cannot make forward progress with counter currents, and anchoring for a time could actually give the fastest route

inverted regions are confusing, but essentially there can be areas that take longer to get to completely surrounded by areas that take less time to get to.   In this case the propagation continues inwards instead of outwards.   This can occur from land (islands, think about sailing back into a bay) or more rarely from weather patterns

Reply:
> anchoring was never very well implemented but the idea is sometimes the boat cannot make forward progress with counter currents, and anchoring for a time could actually give the fastest route

Ah ok, thanks. I'll add code comments.